### PR TITLE
Fixes #12684: Add a package module for cfengine supporting slackware

### PR DESCRIPTION
**Pull request by VictorQrt**
https://www.rudder-project.org/redmine/issues/12684

### CODE REVIEW

File: tree/10_ncf_internals/modules/packages/slackpkg [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+case "${CMD}" in
+    get-package-data)
+        get_package_data
+        ;;
+    list-installed)
+        ls -l /var/log/packages/ |sed -e 's/.* \([^ ]\+\)-\([_a-w0-9.?]\+\)-\(.*\)-.*/Name=\1\'$'\nVersion=\\2\\nArchitecture=\\3/'  
+        ;;
+    list-updates-local|list-updates)
+      slackpkg -dialog=off -batch=on upgrade-all | sed -r 's!^(.+)-(.+)-(i[0-9]86|noarch|x86_64|x86|fw|npmjs)-(.*)$!Name=\1\nVersion=\2\nArchitecture=\3!g' | grep -vE "Total\ package|Do\ you\ wish\ to|Looking\ for|Checking\ local|^$"
```
Comment (amousset) [inline tree/10_ncf_internals/modules/packages/slackpkg]: wrong indentation

File: tree/20_cfe_basics/ncf_lib.cf [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #####################################################################################
 # Copyright 2013 Normation SAS
 #####################################################################################
-#
+# AAAAAAA
```
Comment (amousset) [inline tree/20_cfe_basics/ncf_lib.cf]: https://guadhelpme.files.wordpress.com/2012/01/home_alone_boy1.jpg

File: tree/10_ncf_internals/modules/packages/slackpkg [hunk 3]
```diff
@@ -0,0 +1,188 @@
+#!/bin/sh
+
+CMD="$1"
+
+DEFARCH=`uname -m`
```
Comment (peckpeck) [inline tree/10_ncf_internals/modules/packages/slackpkg]: use $()

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

File: tree/30_generic_methods/package_install_version_cmp_update.cf [hunk 4]
```diff
@@ -165,19 +165,54 @@ bundle agent package_install_version_cmp_update(package_name, version_comparator
                         classes        => classes_generic_two("${old_class_prefix}", "${class_prefix}"),
                         action         => policy("${action}");
```
Comment (amousset) [inline tree/30_generic_methods/package_install_version_cmp_update.cf]: This file should not be modified, we do not implement package management with the deprecated methods.

File: tree/20_cfe_basics/package_lib.cf [hunk 5]
```diff
@@ -140,17 +150,18 @@ bundle agent ncf_package(name, version, architecture, provider, state, options)
       "use_yum_provider" expression => strcmp("yum", "${provider}");
       "use_zypper_provider" expression => strcmp("zypper", "${provider}");
       "use_pkg_provider" expression => strcmp("pkg", "${provider}");
+      "use_slackpkg_provider" expression => strcmp("slackpkg", "${provider}");
       # Used in tests
       "use_test_provider" expression => strcmp("test", "${provider}");
 
     any::
-      "supported_provider" expression => "use_apt_provider|use_yum_provider|use_zypper_provider|use_pkg_provider|use_test_provider";
+      "supported_provider" expression => "use_slackpkg_provider|use_apt_provider|use_yum_provider|use_zypper_provider|use_pkg_provider|use_test_provider";
```
Comment (amousset) [inline tree/20_cfe_basics/package_lib.cf]: Could you try to keep the same order of package managers everywhere?

File: tree/30_generic_methods/file_check_block_present.cf [hunk 6]
```diff
@@ -0,0 +1,47 @@
+#####################################################################################
```
Comment (amousset) [inline tree/30_generic_methods/file_check_block_present.cf]: This should not be here

**Comment by VictorQrt** [comment]
PR rebased

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
PR rebased

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
PR rebased

**Comment by VictorQrt** [comment]
Commit modified

**Comment by VictorQrt** [comment]
Commit modified

File: tree/10_ncf_internals/modules/packages/slackpkg [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+case "${CMD}" in
+    get-package-data)
+        get_package_data
+        ;;
+    list-installed)
+        ls -l /var/log/packages/ |sed -e 's/.* \([^ ]\+\)-\([_a-w0-9.?]\+\)-\(.*\)-.*/Name=\1\'$'\nVersion=\\2\\nArchitecture=\\3/'  
+        ;;
+    list-updates-local|list-updates)
+        slackpkg -dialog=off -batch=on upgrade-all | sed -r 's!^(.+)-(.+)-(i[0-9]86|noarch|x86_64|x86|fw|npmjs)-(.*)$!Name=\1\nVersion=\2\nArchitecture=\3!g' | grep -vE "Total\ package|Do\ you\ wish\ to|Looking\ for|Checking\ local|^$"
```
Comment (amousset) [inline tree/10_ncf_internals/modules/packages/slackpkg]: Could you use a variable for the slack command + all common options?

**Comment by peckpeck** [comment]
OK, merging this PR

**Pull request merged**